samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/qti/XMLImportBean.java

```diff
@@ -461,59 +463,102 @@ public void setAuthorizationBean(AuthorizationBean authorizationBean)
    * Value change on upload
    * @param e the event
    */
-  public void importPoolFromQti(ValueChangeEvent e)
-  {
+  public void importPool(ValueChangeEvent e) {
+    String sourceType = ContextUtil.lookupParam("sourceType");
     String uploadFile = (String) e.getNewValue();
 
-    try
-    {
-    	processAsPoolFile(uploadFile);
+    if (uploadFile!= null && uploadFile.startsWith("SizeTooBig:")) {
+        Long sizeMax = Long.valueOf(ServerConfigurationService.getString("samigo.sizeMax", "20480"));
+        String sizeTooBigMessage = MessageFormat.format(rb.getString("import_size_too_big"), uploadFile.substring(11), Math.round(sizeMax.floatValue()/1024));
+        FacesMessage message = new FacesMessage(sizeTooBigMessage);
+        FacesContext.getCurrentInstance().addMessage(null, message);
+        // remove unsuccessful file
+        log.debug("****Clean up file: {}", uploadFile);
```

Comment:
maybe only output the first so many chars

Reply:
Solved